pkg/kubernetes/kubernetes.go

```diff
@@ -202,6 +202,29 @@ func (h *Client) LabelNodeAsMaster(name string) (err error) {
 	return nil
 }
 
+// WaitUntilReady waits for a node to be ready.
+func (h *Client) WaitUntilReady(name string) error {
+	return retry.Exponential(3*time.Minute, retry.WithUnits(250*time.Millisecond), retry.WithJitter(50*time.Millisecond)).Retry(func() error {
+		attemptCtx, attemptCtxCancel := context.WithTimeout(context.TODO(), 30*time.Second)
+		defer attemptCtxCancel()
+
+		node, err := h.CoreV1().Nodes().Get(attemptCtx, name, metav1.GetOptions{})
+		if err != nil {
+			return retry.UnexpectedError(err)
```

Comment:
I think we need to check if the error is a "Not found" (there is a Kubernetes function for this somewhere) error and make that expected. This was the original error we saw in CI:

```
2020/07/10 00:09:09 error running phase 9 in boot sequence: task 1/1: failed, failed to uncordon node e2e-docker-worker-1: 1 error(s) occurred:
	nodes "e2e-docker-worker-1" not found
```